src/uistate.h

```diff
@@ -8,46 +11,52 @@
 #include <vector>
 #include <string>
 
+class item;
+
 /*
   centralized depot for trivial ui data such as sorting, string_input_popup history, etc.
   To use this, see the ****notes**** below
 */
 class uistatedata : public JsonSerializer, public JsonDeserializer
 {
+    /**** this will set a default value on startup, however to save, see below ****/
+    private:
+        // not needed for compilation, but keeps syntax plugins happy
+        typedef std::string itype_id;
+        enum side { left  = 0, right = 1, NUM_PANES = 2 };
     public:
-        using itype_id = std::string;
         /**** declare your variable here. It can be anything, really *****/
-        int wishitem_selected;
-        int wishmutate_selected;
-        int wishmonster_selected;
-        int iexamine_atm_selected;
-        int adv_inv_leftsort;
-        int adv_inv_rightsort;
-        int adv_inv_leftarea;
-        int adv_inv_rightarea;
-        int adv_inv_leftindex;
-        int adv_inv_rightindex;
-        int adv_inv_src;
-        int adv_inv_dest;
-        int adv_inv_last_popup_dest;
-        int adv_inv_container_location;
-        int adv_inv_container_index;
-        itype_id adv_inv_container_type;
-        itype_id adv_inv_container_content_type;
-        int ags_pay_gas_selected_pump;
-        std::string adv_inv_leftfilter;
-        std::string adv_inv_rightfilter;
-        bool adv_inv_load_veh;
-        int  adv_inv_veh_location_src;
-        int  adv_inv_veh_location_dest;
-
-        bool editmap_nsa_viewmode;        // true: ignore LOS and lighting
-        bool overmap_blinking;            // toggles active blinking of overlays.
-        bool overmap_show_overlays;       // whether overlays are shown or not.
+        int wishitem_selected = 0;
+        int wishmutate_selected = 0;
+        int wishmonster_selected = 0;
+        int iexamine_atm_selected = 0;
+        std::vector<int> adv_inv_sort = {1, 1};
```

Comment:
This (and the other vectors) should probably be `std::array<int, 2>`. I don't expect it to store an arbitrary number of elements, but *always* 2, so why allow resizing it?.